src/dbnode/persist/fs/merger.go

```diff
@@ -126,13 +126,7 @@ func (m *merger) Merge(
 	if err := reader.Open(openOpts); err != nil {
 		return closer, err
 	}
-	defer func() {
-		// Only set the error here if not set by the end of the function, since
-		// all other errors take precedence.
-		if err == nil {
-			err = reader.Close()
-		}
-	}()
+	defer reader.Close() // nolint
```

Comment:
Maybe `err = xerrors.FirstError(err, reader.Close())`, which should both invoke `reader.Close()` and avoid swallowing any errors emitted by it, while keeping previous errors as mentioned in the comment.

Reply:
Reassigning the value of `err` from deferred block is actually a noop, it does not change the return value of the function.